woocommerce/woocommerce-ios
# [Product Bundles] Show product bundle stock status or quantity in product list

Closes: #8954
⚠️ Depends on #9132 


## Description

This updates the stock status and stock quantity displayed for product bundles in product lists, when the feature flag is enabled:

* When the product bundle is in stock and the stock quantity is managed, it displays the product **bundle** stock quantity instead of the product stock quantity.
* When the product bundle does not manage the stock quantity, it displays the product **bundle** stock status instead of the product stock status.
* For any other product type (not product bundles), or when the feature flag is disabled, the product stock quantity or status will be displayed as before.

This is notable in two scenarios:

* If the Product > Inventory settings have e.g. an "In stock" status but one of the bundled products is out of stock, the product list will show "Insufficient stock" as the stock status.
* If the Product > Inventory settings have e.g. 5 in stock but one of the bundled products only has 1 in stock, the product list will show "1 in stock" as the stock quantity.

## Testing instructions


### Prerequisites

1. Install and activate the [Product Bundles extension](https://woocommerce.com/products/product-bundles/) on your store.
2. Create a product with the Product Bundle type.
   *  In the "Bundled products" settings, add a product that is out of stock (with its "Manage stock?" setting unchecked).
   *  In the "Inventory" settings, set the product bundle to in stock (with its "Manage stock?" setting unchecked).
3. Create another product with the Product Bundle type.
   * In the "Bundled products" settings, add a product that has a stock quantity of 1.
   *  In the "Inventory" settings, set the product bundle stock quantity to more than 1.
4. In the wp-admin product list, confirm you see "Insufficient stock" for the first product bundle and "In stock (1)" for the second product bundle.

Example:

![Screenshot 2023-03-14 at 6 54 03 PM](https://user-images.githubusercontent.com/8658164/225108516-9e667b0a-b686-41df-b662-5cbf3b78d350.png)


### To test

1. Build and run the app.
2. Go to the Products tab.
3. Confirm your first product bundle shows "Insufficient stock" in the product list.
4. Confirm your second product bundle shows "1 in stock" in the product list.
5. Go to the Orders tab and create a new order.
6. Select "Add product" and confirm the correct stock status/quantity also appears there (same as above).

## Screenshots


/|Before|After
-|-|-
Products tab|![Simulator Screen Shot - iPhone 14 Pro - 2023-03-14 at 19 00 07](https://user-images.githubusercontent.com/8658164/225109808-e25d33f9-68f1-4d6c-ad15-3cf15da5f8af.png)|![Simulator Screen Shot - iPhone 14 Pro - 2023-03-14 at 18 30 53](https://user-images.githubusercontent.com/8658164/225106520-17cb6f01-ddbd-4db4-ade0-b28969a7e96e.png)
Product selector (e.g. order creation)|![Simulator Screen Shot - iPhone 14 Pro - 2023-03-14 at 19 00 13](https://user-images.githubusercontent.com/8658164/225109818-856d0b60-71be-470d-b2c0-8f49a5a24174.png)|![Simulator Screen Shot - iPhone 14 Pro - 2023-03-14 at 18 31 02](https://user-images.githubusercontent.com/8658164/225106583-4ac6ed6b-66c1-401f-bb2a-3b2374e30326.png)



---
